app/src/main/java/com/example/atheneum/activities/BookInfoActivity.java

```diff
@@ -132,6 +150,79 @@
+                    Barcode barcode = data.getParcelableExtra("Barcode");
+                    String barcodeISBN = String.valueOf(barcode.displayValue);
+
+                    if (barcodeISBN.equals(Long.toString(bOok.getIsbn()))) {
+                        Log.i(TAG, "They do equal");
+
+
+                        TransactionViewModelFactory factory = new TransactionViewModelFactory(bOok.getBookID());
+                        transactionViewModel = ViewModelProviders.of(BookInfoActivity.this, factory).get(TransactionViewModel.class);
+                        final LiveData<Transaction> transactionLiveData = transactionViewModel.getTransactionLiveData();
+
+                        transactionLiveData.observe(this, new Observer<Transaction>() {
+                            @Override
+                            public void onChanged(@Nullable Transaction transaction) {
+                                if (transaction != null) {
+                                    Log.i(TAG, "updateTransaction(): got transaction" + transaction.toString());
+                                    Log.i(TAG, "bookID:  " + String.valueOf(transaction.getBookID()));
+                                    Log.i(TAG, "BScan value:" + String.valueOf(transaction.getBScan()));
+                                    Log.i(TAG, "OScan value:" + String.valueOf(transaction.getOScan()));
+                                    Log.i(TAG, "Owner value:" + transaction.getOwnerID());
+                                    Log.i(TAG, "Borrower value:" + transaction.getBorrowerID());
+                                    transaction.setOScan(true);
+
+
+                                    transaction.setBorrowerID(borrowerID);
+                                    transaction.setOwnerID(loggedInUser.getUserID());
+                                    transactionViewModel.updateTransaction(transaction);
+
+
+                                    if (transaction.getBScan() && transaction.getOScan()) {
+                                        if (transaction.getType().equals("CHECKOUT")) {
+                                            bOok.setStatus(Book.Status.BORROWED);
+                                            DatabaseWriteHelper.updateBook(bOok);
+
+                                            transaction.setBScan(false);
+                                            transaction.setOScan(false);
+                                            transaction.setType(Transaction.RETURN);
+                                            DatabaseWriteHelper.updateTransaction(transaction);
+                                        } else {
+                                            bOok.setStatus(Book.Status.AVAILABLE);
+                                            bOok.setBorrowerID("");
+                                            DatabaseWriteHelper.updateBook(bOok);
+                                            DatabaseWriteHelper.deleteTransaction(transaction);
```

Comment:
What happens to the requests for the book? Should they be deleted? @MarcusBoay could you chime in with your thoughts?

Reply:
I discussed this with Marcus yesterday and it should be deleted but I just haven't gone around to doing it yet

Reply:
It is possible to put everything in the onChanged method into the ViewModel so that the business logic doesn't leak into the activity?

Reply:
I can change the transactions to ViewModel but is it necessary to create a new bookViewModel just to update book status?